pfring/pfring.go

```diff
@@ -222,6 +222,26 @@ func (r *Ring) SetSamplingRate(rate int) error {
 	}
 	return nil
 }
+// SetPollWatermark sets the pfring's poll watermark packet count
+func (r *Ring) SetPollWatermark(count uint16) error {
+	if rv := C.pfring_set_poll_watermark(r.cptr, C.u_int16_t(count)); rv != 0 {
+		return fmt.Errorf("Unable to set poll watermark, got error code %d", rv)
+	}
+	return nil
+}
+
+// SetPriority sets the pfring poll threads CPU usage limit
+func (r *Ring) SetPriority(cpu uint16) {
+	C.pfring_config(C.u_short(cpu))
+}
+
+// SetPollDuration sets the pfring's poll duration before it yields/returns
+func (r *Ring) SetPollDuration(duration uint) error {
```

Comment:
would we be able to pass in a time.Duration here and convert it into a u_int under the hood?

Reply:
I have just checked time.Duration and it is uint64 type representing nano second difference. pfring_set_poll_duration treats the value passed in as time in milliseconds. So there is no benefit in passing nano seconds and then immediately converting the value to milli seconds. Please let me know if I did not understand your point completely.

Reply:
The point is simply that, in Go world, there is a canonical type for Duration, and we should use that where possible.  My preference would be something like:

func (r *Ring) SetPollDuration(d time.Duration) {
  millis := <convert d to millis>
  ...call with millis...
}

Reply:
Alternatively, if you're wedded to passing in an integer of some sort, please rename the passed-in arg to durationMillis or something like that that makes it clear what time unit you're expecting.

Reply:
Yeah, I'll do that. Shall submit new code soon.